tnz/zti.py

```diff
@@ -2694,6 +2702,55 @@ def __refresh(self):
 
         self.stdscr.refresh(_win_callback=self.__set_event_fn())
 
+    def __register_macros(self):
+        import importlib.util
+        import sys
+        import types
+
+        if not os.path.isdir(self.macros_dir):
+            _logger.exception(f"{self.macros_dir} is not a directory")
+            return
+
+        for macro_file in os.listdir(self.macros_dir):
+            if not macro_file.endswith(".py"):
+                continue
+
+            if len(macro_file.split('.')) > 1:
+                continue
+
+            macro_name = macro_file.split('.')[0]
```

Comment:
Since `precmd()` converts commands to lowercase, any command with a capital letter will not work. Along those lines... also consider ignoring/rejecting macros that have a space in the name.